sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads_async.py

```diff
@@ -224,16 +226,16 @@ def get_substream_blocks(self):
         for i in range(blocks):
             index = i * self.chunk_size
             length = last_block_size if i == blocks - 1 else self.chunk_size
-            yield ('BlockId{}'.format("%05d" % i), SubStream(self.stream, index, length, lock))
+            yield index, SubStream(self.stream, index, length, lock)
```

Comment:
if I remember properly blockid should have the same length, if we change it to index I'm worried something bad will happen.

Reply:
I did not use the index, I still converted it to a proper block id here https://github.com/Azure/azure-sdk-for-python/pull/17272/files#diff-fd90426e10c45bf19310c14aef222c23145b307e761fef45eac642e704562103R269
I just moved it so this method can be used by datalake as well as blob

Reply:
found the reference here https://docs.microsoft.com/en-us/rest/api/storageservices/put-block#uri-parameters

Reply:
nvm! I found you formated the blockid in another place!